Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -47,25 +47,34 @@ extension NavigationMapView {
     }
     
     /**
-     Update the existing primary route line during active navigation.
-     
-     - parameter routeProgress: The current `RouteProgress`.
-     - parameter coordinate: The current user location coordinate.
+     Update the route line when route progress did change during active navigation.
      */
-    func updateRouteLine(routeProgress: RouteProgress, coordinate: CLLocationCoordinate2D?) {
-        show([routeProgress.route], legIndex: routeProgress.legIndex)
+    func updateRouteLineWithProgress(routeProgress: RouteProgress, coordinate: CLLocationCoordinate2D?) {
         guard routeLineTracksTraversal else { return }
-        
         if routeProgress.routeIsComplete {
             removeRoutes()
         } else {
             updateUpcomingRoutePointIndex(routeProgress: routeProgress)
-            offRouteDistanceCheckEnabled = false
             travelAlongRouteLine(to: coordinate)
-            offRouteDistanceCheckEnabled = true
         }
     }
     
+    /**
+     Redraw the primary route line during active navigation when refresh or reroute event occurs.
+     
+     - parameter routeProgress: The current `RouteProgress`.
+     - parameter coordinate: The current user location coordinate.
+     */
+    public func redrawRouteLineWithProgress(routeProgress: RouteProgress, coordinate: CLLocationCoordinate2D?) {
```

Comment:
Looks like we've covered cases for redrawing line on refreshing and rerouting, what other use-cases do you see where this public method would be required and actually called by user? Maybe we could leave this one as `internal` instead?